src/client/components/JoinForm/JoinForm.jsx

```diff
@@ -0,0 +1,75 @@
+import { useForm } from 'react-hook-form';
+import styles from './join-form.module.css';
+
+export const JoinForm = ({ connect }) => {
+  const { handleSubmit, register } = useForm();
+
+  const onSubmit = (data) => {
+    localStorage.setItem('username', data.username);
+    connect(data);
+  };
+
+  const savedUsername = localStorage.getItem('username') || '';
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className={styles['join-form']}>
+      <div className={styles['join-form__field-wrapper']}>
+        <label htmlFor="username">Username:</label>
+        <input
+          type="text"
+          id="username"
+          placeholder="Enter your username"
+          defaultValue={savedUsername}
+          {...register('username', { required: true, minLength: 2 })}
+          className={styles['join-form__field']}
+        />
+      </div>
+      <div className={styles['join-form__field-wrapper']}>
+        <label htmlFor="room">Room name:</label>
```

Comment:
Similar to the username input, consider adding an `aria-label` or `aria-describedby` attribute to the room input for improved accessibility.